src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -565,22 +565,6 @@ sealed private[immutable] trait LazyListFactory[+CC[+X] <: LinearSeq[X] with Laz
     */
   def continually[A](elem: => A): CC[A] = newCons(elem, continually(elem))
 
-  /**
-    * @return a LazyList by using a function `f` producing elements of
-    *         type `A` and updating an internal state `S`.
-    * @param init State initial value
-    * @param f    Computes the next element (or returns `None` to signal
-    *             the end of the collection)
-    * @tparam A   Type of the elements
-    * @tparam S   Type of the internal state
-    */
-  def unfold[A, S](init: S)(f: S => Option[(A, S)]): CC[A] = {
```

Comment:
Is the new generic implementation as efficient as this one and sufficiently lazy?

Reply:
it certainly should be sufficiently lazy, since it just creates a `LazyList` from a `View`. I can't speak to its performance (I haven't done any benchmarks), but I wouldn't imagine it to be substantially different.